waku/v2/utils/peers.nim

```diff
@@ -112,7 +112,49 @@ func getTransportProtocol(typedR: TypedRecord): Option[IpTransportProtocol] =
 
 ## Parses a fully qualified peer multiaddr, in the
 ## format `(ip4|ip6)/tcp/p2p`, into dialable PeerInfo
-proc parseRemotePeerInfo*(address: string): RemotePeerInfo {.raises: [Defect, ValueError, LPError].}=
+proc parsePeerInfo*(address: string):
+                    Result[RemotePeerInfo, string] =
+  var
+    nwPart, tcpPart, p2pPart, wsPart, wssPart: MultiAddress
+
+  try:
+    let multiAddr = MultiAddress.init(address).tryGet()
+    for addrPart in multiAddr.items():
+      case addrPart[].protoName()[]
+      # All protocols listed here: https://github.com/multiformats/multiaddr/blob/b746a7d014e825221cc3aea6e57a92d78419990f/protocols.csv
+      of "ip4", "ip6", "dns", "dnsaddr", "dns4", "dns6":
+        nwPart = addrPart.tryGet()
+      of "tcp":
+        tcpPart = addrPart.tryGet()
+      of "p2p":
+        p2pPart = addrPart.tryGet()
+      of "ws":
+        wsPart = addrPart.tryGet()
+      of "wss":
+        wssPart = addrPart.tryGet()
+
+    # nim-libp2p dialing requires remote peers to be initialised with a peerId and a wire address
+    let
+      peerIdStr = p2pPart.toString()[].split("/")[^1]
+      wireAddr = nwPart & tcpPart & wsPart & wssPart
+
+    if (not wireAddr.validWireAddr()):
+      return err("Invalid node multi-address")
```

Comment:
```suggestion
      return err("Invalid node multiaddress")
```

Reply:
Thanks for the suggestions @rymnc ! I've applied them.